### PR TITLE
[Sign Up] User have an error on sign up when enter 1 letter name

### PR DESCRIPTION
**Pull request by ABbondar**
# GreenCityUser PR
https://github.com/ita-social-projects/GreenCity/issues/5206

## Summary Of Changes :fire:
Modified regexp pattern on OwnSignUpDto "name" field

## Issue Link :clipboard:
_&lt;Link to the issue&gt;_

## Added
Added tests for regexp patterns

## Changed
* _&lt;detail item of what was changed&gt;_
* _&lt;describe functionality that was changed&gt;_

## Deleted
* _&lt;detail item of what was removed&gt;_

## How to test :clipboard:
Running tests in OwnSignUpDtoTest class

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

**Review by LiliaMokhnatska — APPROVED** [review]

**Review by Markiro1 — APPROVED** [review]

**Review by juseti — APPROVED** [review]

File: service-api/src/main/java/greencity/security/dto/ownsecurity/OwnSignUpDto.java [hunk 1]
```diff
@@ -22,12 +21,8 @@
...
 @Builder
 public class OwnSignUpDto {
-    @NotBlank
-    @Length(
-        min = ValidationConstants.USERNAME_MIN_LENGTH,
-        max = ValidationConstants.USERNAME_MAX_LENGTH)
     @Pattern(
-        regexp = "^[a-zA-Z0-9]+([._]?[a-zA-Z0-9]+)++$",
+        regexp = ValidationConstants.USERNAME_REGEXP,
         message = ValidationConstants.INVALID_USERNAME)
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/security/dto/ownsecurity/OwnSignUpDto.java]: What is the value of `ValidationConstants.INVALID_USERNAME`
Should we also update it too?

Comment (ABbondar) [reply service-api/src/main/java/greencity/security/dto/ownsecurity/OwnSignUpDto.java]: updated

File: service-api/src/main/java/greencity/security/dto/ownsecurity/OwnSignUpDto.java [hunk 2]
```diff
@@ -22,13 +21,9 @@
 @NoArgsConstructor
 @Builder
 public class OwnSignUpDto {
-    @NotBlank
-    @Length(
-        min = ValidationConstants.USERNAME_MIN_LENGTH,
-        max = ValidationConstants.USERNAME_MAX_LENGTH)
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/security/dto/ownsecurity/OwnSignUpDto.java:28]: I see that ` UserForListDto, UserManagementDto, UserManagementUpdateDto, UserUpdateDto, EmployeeSignUpDto` use the same validation  
   ```
    min = ValidationConstants.USERNAME_MIN_LENGTH,
    max = ValidationConstants.USERNAME_MAX_LENGTH)
```
Should it also be changed to `regexp = ValidationConstants.USERNAME_REGEXP`?

**Review by RomaMocherniuk — CHANGES_REQUESTED** [review]
Left a comment

**Review by RomaMocherniuk — APPROVED** [review]

**Pull request merged**